tests/test_error_handling.py

```
@@ -0,0 +1,596 @@
+        """Test server tools with invalid parameters."""
+        mock_service = Mock()
+        mock_service.search_tracks = AsyncMock(return_value=[])
+        
+        with patch('tidal_mcp.server.ensure_service', return_value=mock_service):
+            # Test with empty track list
+            result = await tidal_add_to_playlist("playlist123", [])
+            assert result["success"] is False
+            assert "No track IDs provided" in result["error"]
+            
+            # Test with empty indices list
+            result = await tidal_remove_from_playlist("playlist123", [])
```

Ruff (0.12.2)

Undefined name `tidal_remove_from_playlist`

(F821)